File: 백준/Gold/13164. 행복 유치원/행복 유치원.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
#가장 키가 큰 원생과 가장 키가 작은 원생의 키 차이만큼
#K개의 조에 대해 티셔츠 만드는 비용의 합을 최소
n, k = map(int, input().split())
person = list(map(int, input().split()))
diffs = []
for i in range(1, n):
    diffs.append(person[i]-person[i-1])
diffs.sort(reverse=True)
if k==n:
    print(0)
else:
    print(sum(diffs[k-1:]))

File: 백준/Gold/13164. 행복 유치원/행복 유치원.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var (
	reader = bufio.NewReader(os.Stdin)
	writer = bufio.NewWriter(os.Stdout)
)

func main() {
	defer writer.Flush()
	var n, k, res int
	fmt.Fscanf(reader, "%d %d\n", &n, &k)
	var person = make([]int, n)
	var diff = make([]int, n-1)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &person[i])
	}
	for i := 1; i < n; i++ {
		diff[i-1] = person[i] - person[i-1]
	}
	sort.Sort(sort.Reverse(sort.IntSlice(diff)))
	if n == k {
		fmt.Fprintf(writer, "%d\n", 0)
	} else {
		for i := k - 1; i < len(diff); i++ {
			res += diff[i]
		}
		fmt.Fprintf(writer, "%d\n", res)
	}
}
